lib/data_table.rb

```diff
@@ -881,6 +881,24 @@ def to_csv(view_context, params, options)
 
         cols = @columns.select{ |col| col.downloadable? }
 
+        if cols.present?
+          cols.each do |column|
+            user_column = options[:columns_layout]['columns'].detect do |c|
+              col_name = column.name.to_s
+              split_layout_name = c['dataField'].split('.')
+
+              # This has to be done to handle case where the columns name is an array of symbols instead of a symbol
+              name = split_layout_name.length > 2 ? split_layout_name.last(2).map!(&:to_sym).to_s : split_layout_name.last.to_s
```

Comment:
For the "Overridden By" column, it had @name=[:overridden_by, :first_name], whereas other columns had @name=:entity (as an example). 

Then in options[:column_layout], it's corresponding entry had dataField="<something>.overriden_by.first_name" and hence dataField.split.last would return first_name.

So this is what I came up with and it works for my case, but should probably test in other parts of the system.